src/cmd/build.rs

```diff
@@ -298,8 +295,88 @@ fn exec_cargo_for_wasm_target(
+
+    let tmp_dir = tempfile::Builder::new()
+        .prefix("cargo-contract-dylint_")
+        .tempdir()?;
+    log::debug!("Using temp workspace at '{}'", tmp_dir.path().display());
+
+    let template = include_bytes!(concat!(env!("OUT_DIR"), "/ink-dylint-driver.zip"));
+    crate::util::unzip(template, tmp_dir.path().to_path_buf(), None)?;
+
+    let manifest_path = crate_metadata.manifest_path.cargo_arg();
+    let args = vec!["ink_linting", &manifest_path];
+    let tmp_dir_path = tmp_dir.path().as_os_str().to_string_lossy();
+    let env = vec![
+        ("DYLINT_LIBRARY_PATH", tmp_dir_path.as_ref()),
+        ("DYLINT_DRIVER_PATH", tmp_dir_path.as_ref()),
+    ];
+    let working_dir = crate_metadata
+        .manifest_path
+        .directory()
+        .unwrap_or_else(|| Path::new("."))
+        .canonicalize()?;
+    util::invoke_cargo("dylint", &args, Some(working_dir), verbosity, Some(env))?;
+
+    Ok(())
+}
+
+/// Checks if all requirements for `dylint` are installed.
+///
+/// Those are an installed version of `cargo-dylint` and `dylint-link`.
+fn check_dylint_requirements(_working_dir: &Path) -> Result<()> {
+    let execute_cmd = |cmd: &mut Command| {
+        // when testing this function we set the `PATH` to the `working_dir`
+        // so that we can have mocked binaries in there which are executed
+        // instead of the real ones.
+        #[cfg(test)]
+        {
+            let path_env = std::env::var("PATH").unwrap();
+            let path_env = format!("{}:{}", _working_dir.to_string_lossy(), path_env);
+            cmd.env("PATH", path_env);
+        }
+
+        cmd.stdout(std::process::Stdio::null())
+            .stderr(std::process::Stdio::null())
+            .spawn()
+            .unwrap_or_else(|_| panic!("Error executing `{:?}`", cmd))
+            .wait()
+            .map(|res| res.success())
+            .unwrap_or_else(|err| panic!("Error executing `{:?}`: {:?}", cmd, err))
+    };
+
+    // when testing this function we should never fall back to a `cargo` specified
+    // in the env variable, as this would mess with the mocked binaries.
+    #[cfg(not(test))]
+    let cargo = std::env::var("CARGO").unwrap_or_else(|_| "cargo".to_string());
+    #[cfg(test)]
+    let cargo = "cargo";
+
+    if !execute_cmd(Command::new(cargo).arg("dylint").arg("--version")) {
+        anyhow::bail!("cargo-dylint was not found!\n\
+            Make sure it is installed and the binary is in your PATH environment.\n\n\
+            You can install it by executing `cargo install cargo-dylint`."
+            .to_string()
+            .bright_yellow());
+    }
+
+    if !execute_cmd(Command::new("dylint-link").arg("--version")) {
+        anyhow::bail!("dylint-link was not found!\n\
+            Make sure it is installed and the binary is in your PATH environment.\n\n\
+            You can install it by executing `cargo install dylint-link`."
+            .to_string()
+            .bright_yellow());
```

Comment:
I think this is useless here? We don't need `dylint-link` when running `cargo-contract` but when building it. This should go into `./build.rs` in order to give a proper error when this isn't installed during build.